HopsanGUI/Widgets/TextEditorWidget.cpp

```diff
@@ -0,0 +1,682 @@
+/*-----------------------------------------------------------------------------
+
+ Copyright 2017 Hopsan Group
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ The full license is available in the file GPLv3.
+ For details about the 'Hopsan Group' or information about Authors and
+ Contributors see the HOPSANGROUP and AUTHORS files that are located in
+ the Hopsan source code root directory.
+
+-----------------------------------------------------------------------------*/
+
+//!
+//! @file   ScriptEditor.h
+//! @author Robert Braun <robert.braun@liu.se>
+//! @date   2018-10-01
+//! @version $Id$
+//!
+//! @brief Contains the script editor class
+//!
+
+
+#include "TextEditorWidget.h"
+#include "Utilities/HighlightingUtilities.h"
+#include "global.h"
+#include "Configuration.h"
+#include "MessageHandler.h"
+#include "Widgets/ProjectTabWidget.h"
+#include "HcomHandler.h"
+#include "Widgets/HcomWidget.h"
+
+#include <QGridLayout>
+#include <QFileDialog>
+#include <QStringListModel>
+#include <QAbstractItemView>
+#include <QScrollBar>
+#include <QTextDocumentFragment>
+#include <QKeyEvent>
+#include <QPrinter>
+#include <QPainter>
+#include <QPrintDialog>
+#include <QMimeData>
+#include <math.h>
+
+TextEditorWidget::TextEditorWidget(QFileInfo scriptFileInfo, HighlighterTypeEnum highlighter, QWidget *parent) : QWidget(parent)
+{
+    mFileInfo = scriptFileInfo;
+
+    mpEditor = new TextEditor(highlighter, this);
+    QFont font("Monospace");
+    font.setStyleHint(QFont::TypeWriter);
+    mpEditor->setFont(font);
+
+    mpHcomHighlighter = new HcomHighlighter(nullptr);
+    mpXmlHighlighter = new XmlHighlighter(nullptr);
```

Comment:
Are these ever deleted?
Should use std::unique_ptr instead of raw pointer to solve that automatically

Reply:
In that case I believe it is easier to just delete them in the destructor of the parent class.